var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -76,6 +76,43 @@ class Trilinos(CMakePackage):
     variant('xsdkflags',    default=False,
             description='Compile using the default xSDK configuration')
 
+    # CUDA options (compare to kokkos/package.py)
+    variant('cuda', default=False, description="enable Cuda backend")
+    # kokkos has conflicts with cuda for kokkos@2.5:2.7, restrict this variant
+    # to @12.14.1 or later.
+    conflicts('+cuda', when='@:12.14.0')
+    conflicts('+cuda', when='~kokkos')
+    variant('kokkos_enable_cuda_uvm', default=False,
```

Comment:
Does it make sense to set it to `False` when Tpetra requires it?

Reply:
I don't want this variant to default to `True` for `~cuda` builds.  However, I will add the following to cover your UseCase.

```
conflicts('~kokkos_enable_cuda_lambda', when='+cuda+tpetra')
conflicts('~kokkos_enable_cuda_uvm', when='+cuda+tpetra')
```